python/src/trezorlib/nostr.py

```diff
@@ -0,0 +1,62 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2025 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+
+import json
+from typing import TYPE_CHECKING, AnyStr
+
+from . import messages
+from .tools import expect
+
+if TYPE_CHECKING:
+    from .client import TrezorClient
+    from .protobuf import MessageType
+    from .tools import Address
+
+
+@expect(messages.NostrPubkey)
+def get_pubkey(
+    client: "TrezorClient",
+    n: "Address",
+) -> "MessageType":
+    return client.call(
+        messages.NostrGetPubkey(
+            address_n=n,
+        )
+    )
+
+
+@expect(messages.NostrEventSignature)
+def sign_event(
+    client: "TrezorClient",
+    n: "Address",
+    event: AnyStr,
+) -> "MessageType":
+    event_json = json.loads(event)
```

Comment:
if you want a structured dict in, pass it in as a dict, not as a string. Let the caller do their own `json.loads` if they want to. for bonus points, use `TypedDict`.
(wondering if we should instead introduce a dataclass for the parameters, or perhaps use the protobuf message directly? given that we're _returning_ the corresponding protobuf instance too)